From f8055c4111175d284e1d7ed7de4f4c73d1e65331 Mon Sep 17 00:00:00 2001
From: Joel Holdsworth <joel@airwebreathe.org.uk>
Date: Mon, 11 May 2026 09:45:16 -0700
Subject: [PATCH 13/15] mountmgr: Report FILE_SUPPORTS_OPEN_BY_FILE_ID.

The mountmgr volume device path should also report FILE_SUPPORTS_OPEN_BY_FILE_ID
in the filesystem attribute flags for NTFS volumes, matching the behavior of real
Windows NTFS volumes.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=59729
---
 dlls/kernel32/tests/volume.c | 2 +-
 dlls/mountmgr.sys/device.c   | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/kernel32/tests/volume.c b/dlls/kernel32/tests/volume.c
index 1ec21a8a6e..8b7a705e19 100644
--- a/dlls/kernel32/tests/volume.c
+++ b/dlls/kernel32/tests/volume.c
@@ -2019,7 +2019,7 @@ static void test_GetVolumeInformationByHandle(void)
     ret = pGetVolumeInformationByHandleW( file, label, ARRAY_SIZE(label), &serial,
             &filename_len, &flags, fsname, ARRAY_SIZE(fsname) );
     ok(ret, "got error %lu\n", GetLastError());
-    todo_wine ok(flags & FILE_SUPPORTS_OPEN_BY_FILE_ID,
+    ok(flags & FILE_SUPPORTS_OPEN_BY_FILE_ID,
             "expected FILE_SUPPORTS_OPEN_BY_FILE_ID to be set, got %#lx\n", flags);
 
     memset(buffer, 0, sizeof(buffer));
diff --git a/dlls/mountmgr.sys/device.c b/dlls/mountmgr.sys/device.c
index 5ae4220931..54d32ad2f1 100644
--- a/dlls/mountmgr.sys/device.c
+++ b/dlls/mountmgr.sys/device.c
@@ -1756,7 +1756,8 @@ static NTSTATUS WINAPI harddisk_query_volume( DEVICE_OBJECT *device, IRP *irp )
             break;
         default:
             fsname = L"NTFS";
-            info->FileSystemAttributes = FILE_CASE_PRESERVED_NAMES | FILE_PERSISTENT_ACLS | FILE_SUPPORTS_REPARSE_POINTS;
+            info->FileSystemAttributes = FILE_CASE_PRESERVED_NAMES | FILE_PERSISTENT_ACLS |
+                                         FILE_SUPPORTS_REPARSE_POINTS | FILE_SUPPORTS_OPEN_BY_FILE_ID;
             info->MaximumComponentNameLength = 255;
             break;
         }
-- 
2.53.0

